docs/DeveloperGuide.adoc

```diff
@@ -349,6 +349,57 @@ We are using `java.util.logging` package for logging. The `LogsCenter` class is
 
 Certain properties of the application can be controlled (e.g user prefs file location, logging level) through the configuration file (default: `config.json`).
 
+=== Link Management Feature
+
+The link management feature allows users to add and manage a set of website links that will be associated with a given module.
+
+==== Implementation
+
+The link management feature is facilitated by the `link` class.
+
+Each `link` object is stored in an `ArrayList<Link>`, which in turn is stored in its corresponding `TrackedModule`.
+
+The link feature currently supports add, delete, edit and launch commands.
+
+An example usage scenario of an *add link* operation is given below and the behavior of the link management feature is shown.
+
+1. The user launches the application with an existing save file. `ModuleBook` is initialized with existing `TrackedModule`.
+
+2. The user inputs `link MODULE_INDEX a/add n/LINK_NAME l/LINK_URL` to add a link with name `LINK_NAME`, linked to `LINK_URL` to the `TrackedModule` with the corresponding `MODULE_INDEX`.
+
+3. `LogicManager` receives user input and parses it using `ModuleBookParser#parseCommand()`.  `ModuleBookParser` reads the `COMMAND_WORD` and identifies the input as a `Link` related command and passes the input to `LinkCommandParser`.
+
+4. `LinkCommandParser` determines the `add` action required through the `a/` prefix argument. It then proceeds to pass the relevant input to  `AddLinkCommandParser`.
+
+5. `AddLinkCommandParser` checks for the validity of `LINK_URL` given by the user and creates the relevant `Link` and `AddLinkCommand` objects.
+
+6. `AddLinkCommand` calls `Model#get(MODULE_INDEX)` to find the required module. If it does not exist or is not currently tracked, `AddLinkCommand` throws a `CommandException`. Otherwise, it then proceeds to call `TrackedModule#hasLinkTitle(LINK_NAME)` to check for an existing `Link` object with the same `LINK_NAME`.
+
+7. If such a `Link` object exists, `AddLinkCommand` aborts and throws a `CommandException`. Otherwise, the created `Link` will be added to the specified `TrackedModule`.
+
+8. If the operation is successful, a `CommandResult` with the success message will be returned.
+
+.Sequence diagram for add link operation
+image::AddLinkCommandSequenceDiagram.png[]
```

Comment:
Would you like to consider simplifying the diagram so it isn't so wide. 

Because it is so wide, the image is downscaled in the website, which makes the text in the image very small compared to the website font, and more difficult to read.

![image](https://user-images.githubusercontent.com/43642522/67452266-3a2a1400-f655-11e9-8651-311d71ac8c3a.png)